src/cljs/athens/views/athena.cljs

```diff
@@ -273,76 +229,130 @@
-                                                                                               (navigate-uid uid)))
-                                                                                 :class    (when (= i index) "selected")})
-
-                                                   [:div (use-style result-body-style)
-                                                    [:h4.title (use-sub-style result-style :title)
-                                                     [:b "Create Page: "]
-                                                     query]]
-                                                   [:span.link-leader (use-sub-style result-style :link-leader) [(r/adapt-react-class Create)]]]
-
-                                                  [:div (use-style result-style {:key      i
-                                                                                 :on-click (fn []
-                                                                                             (let [selected-page {:node/title   title
-                                                                                                                  :block/uid    uid
-                                                                                                                  :block/string string
-                                                                                                                  :query        query}]
-                                                                                               (dispatch [:athena/toggle])
-                                                                                               (dispatch [:athena/update-recent-items selected-page])
-                                                                                               (navigate-uid uid)))
-                                                                                 :class    (when (= i index) "selected")})
-                                                   [:div (use-style result-body-style)
-
-                                                    [:h4.title (use-sub-style result-style :title) (highlight-match query title)]
-                                                    (when string
-                                                      [:span.preview (use-sub-style result-style :preview) (highlight-match query string)])]
-                                                   [:span.link-leader (use-sub-style result-style :link-leader) [(r/adapt-react-class ArrowForward)]]])))]))]])))})))
+(def track-click-outside-athena
+  (partial view-utils/track-outside-click
+    #(dispatch [:athena/toggle])))
+
+(defn create-page
+  [title openOnSidebar]
+  (let [uid (gen-block-uid)]
+    (dispatch [:athena/toggle])
+    (dispatch [:page/create title uid])
+    (if openOnSidebar
+      (js/setTimeout #(dispatch [:right-sidebar/open-item uid]) 500)
```

Comment:
Maybe this could be one of the side effects of `:page/create`?
This way we wouldn't have to artificially slow down ux 🚀